docs/API-functions.md

```diff
@@ -1089,6 +1090,61 @@ of error has occurred on the socket.
 This is reported only if `msTimeOut` was \>=0, otherwise the function waits
 indefinitely.
 
+### srt_epoll_uwait
```

Comment:
Suggested changes to lines 1093-1147:

--BEGIN--


### srt_epoll_uwait
```
int srt_epoll_uwait(int eid, SRT_EPOLL_EVENT* fdsSet, int fdsSize, int64_t msTimeOut, int edgeMode/* = false*/);
```

This function only reports user socket (SRT socket) events. It blocks a call 
until any readiness state occurs in the epoll container. Keep in mind that the 
readiness states reported in epoll can be **level-triggered or edge-triggered**, 
as specified by the `edgeMode` argument. 

In **edge-triggered** mode the function only returns socket states
which have changed since the last call. On the other hand, **level-triggered**
mode will keep on returning while the state remains the same.

This function blocks until a specified timeout. There are two special timeout values:

- If the timeout is 0, the function exits immediately after checking. 
- If timeout is -1, it blocks indefinitely until a readiness state occurs.

* `eid`: epoll container
* `fdsSet` : A pointer to an array of `SRT_EPOLL_EVENT`
* `fdsSize` : The size of the fdsSet array
* `msTimeOut` : Timeout specified in milliseconds, or special values (0 or -1)
* `edgeMode` : The triggered mode (default is 0 for **level-triggered**)

The `SRT_MSGCTRL` structure:

```
typedef struct SRT_EPOLL_EVENT_
{
	SRTSOCKET fd;
	int       events; // UDT_EPOLL_IN | UDT_EPOLL_OUT | UDT_EPOLL_ERR
} SRT_EPOLL_EVENT;
```

* `fd` : the user socket (SRT socket)
* `events` : any combination of `SRT_EPOLL_IN`, `SRT_EPOLL_OUT` and `SRT_EPOLL_ERR`

Note that when `SRT_EPOLL_ERR` is set the error cannot be retrieved
with `srt_getlasterror` because it means the socket is closed. The 
socket state can be read using `srt_getsockstate`.

- Returns:

  * The total number of user socket (SRT socket) states changed. Note that this
can be larger than `fdsSize`. If `edgeMode` is true, only `fdsSize`
events are returned in the `fdsSet` and the remaining events can be retrieved by
a new call to this function.

  * -1 in case of error

- Errors:

  * `SRT_EINVPOLLID`: `eid` designates no valid EID object
  * `SRT_ETIMEOUT`: Up to `msTimeOut` no sockets subscribed in `eid` were ready.
This is reported only if `msTimeOut` was \>=0. Otherwise the function waits
indefinitely.


--END--

Reply:
Please re-review, there were some more updates

Reply:
When does the function return `SRT_ETIMEOUT`?
The following test fails on `ASSERT_EQ(srt_epoll_uwait(epoll_id, events, 2, 500), SRT_ETIMEOUT);`
`srt_epoll_uwait(...)` returns 0.
```
TEST(CEPoll, UWaitEnableEmpty)
{
    ASSERT_EQ(srt_startup(), 0);

    const int epoll_id = srt_epoll_create();
    ASSERT_GE(epoll_id, 0);

    srt_epoll_set(epoll_id, SRT_EPOLL_ENABLE_EMPTY);

    SRT_EPOLL_EVENT events[2];
    ASSERT_EQ(srt_epoll_uwait(epoll_id, events, 2, 500), SRT_ETIMEOUT);

    try
    {
        EXPECT_EQ(srt_epoll_release(epoll_id), 0);
    }
    catch (CUDTException& ex)
    {
        cerr << ex.getErrorMessage() << endl;
        EXPECT_EQ(0, 1);
    }

    EXPECT_EQ(srt_cleanup(), 0);
}
```